package/MDAnalysis/core/groups.py

```diff
@@ -255,6 +255,22 @@ def setter(self, values):
                     property(getter, setter, None, attr.singledoc))
             cls._SETATTR_WHITELIST.add(attr.singular)
 
+    @classmethod
+    def _del_prop(cls, attr):
+        """Remove `attr` from the namespace for this class
+
+        Parameters
+        ----------
+        attr : A :class:`TopologyAttr` object
+        """
+        if hasattr(cls, attr.attrname):
```

Comment:
Consider
```
try:
    delattr(cls, attr.attrname)
except AttributeError:
    pass
else:
    cls._SETATTR_WHITELIST.discard(attr.attrname)
```

This is marginally safer because it removes the redundancy between `hasattr` and `delattr`.